### ISSUE TITLE
TypeError in Time Dimension Details crashes the app

### ISSUE THREAD

**Issue by ericpgreen2** [rilldata/rill#3989]
<img width="539" alt="image" src="https://github.com/rilldata/rill/assets/14206386/56f0981b-81e1-44e1-bf40-6d59291dfd2b">

**Comment by bcolloran** [comment]
part of #3679

**Comment by bcolloran** [comment]
the crash is in transposeArray:

```js
export function transposeArray(arr, rowCount, columnCount) {
  const columnarBody = [];

  for (let i = 0; i < columnCount; i++) {
    const column = [];
    for (let j = 0; j < rowCount; j++) {
      column.push(arr[j][i]);   // ------------- CRASHES HERE------------------
    }
    columnarBody.push(column);
  }

  return columnarBody;
}
```

there is a stop gap solution in https://github.com/rilldata/rill/pull/4135, but @djbarnwal will need to do type cleanups in the surrounding files to get a more robust behavior. Reassigning to @djbarnwal for a complete fix.

**Comment by bcolloran** [comment]
possibly closed by #4042 and @djbarnwal has additional work that might help in https://github.com/rilldata/rill/pull/4154. Going to close this since we can't directly reproduce, can always re-open if this error code shows up in telemetry again
